src/main/c/netty_quic_boringssl.c

```diff
@@ -564,11 +568,38 @@ int quic_tlsext_servername_callback(SSL *ssl, int *out_alert, void *arg) {
     return resultValue;
 }
 
-static jlong netty_boringssl_SSLContext_new0(JNIEnv* env, jclass clazz, jboolean server, jbyteArray alpn_protos, jobject handshakeCompleteCallback, jobject certificateCallback, jobject verifyCallback, jobject servernameCallback, jint verifyMode, jobjectArray subjectNames) {
+// see https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_set_keylog_callback.html
+void keylog_callback(const SSL* ssl, const char* line) {
+    SSL_CTX* ctx = SSL_get_SSL_CTX(ssl);
+    if (ctx == NULL) {
+        return;
+    }
+
+    JNIEnv* e = NULL;
+    if (quic_get_java_env(&e) != JNI_OK) {
+        return;
+    }
+
+    jobject keylogCallback = SSL_CTX_get_ex_data(ctx, keylogCallbackIdx);
+    if (keylogCallback == NULL) {
+        return;
+    }
+
+    jstring keyString = NULL;
+    if (line != NULL) {
+        keyString = (*e)->NewStringUTF(e, line);
```

Comment:
you should check if `keyString` is `NULL` after this call which might be the case if we did see an OOME